sessiontxn/isolation/optimistic.go

```diff
@@ -89,10 +90,28 @@ func isOptimisticTxnRetryable(sessVars *variable.SessionVars, tp sessiontxn.Ente
 	return false
 }
 
+// GetStmtReadTS returns the read timestamp used by select statement (not for select ... for update)
+func (p *OptimisticTxnContextProvider) GetStmtReadTS() (uint64, error) {
+	// If `math.MaxUint64` is used for point get optimization, it is not necessary to activate the txn.
+	// Just return `math.MaxUint64` to save the performance.
+	if p.optimizeWithMaxTS {
```

Comment:
Notice that we should guarantee that the outer code should NOT make an assumption that the txn is activated after `GetStmtReadTS` when we are using the maxTS optimization for point get.